chapter/1/rpc-reviewed-dont-keep.md

```diff
@@ -0,0 +1,619 @@
+also undergone various modifications as internet standards were set and TCP/IP
+became the norm.
+
+A new breed of RPC also started in this era(early 2000's), Async RPC, giving
+rise to systems that use *futures* and *promises*, like Finagle{% cite finagle
+--file rpc %} and Cap'n Proto(post-2010).
+
+In the post-2000 era, MAUI{% cite maui --file rpc %}, Cap'n Proto{% cite
+capnproto --file rpc %}, gRPC{% cite grpc --file rpc %}, Thrift{% cite thrift
+--file rpc %} and Finagle{% cite finagle --file rpc %} have been released, which
+have significantly boosted the widespread use of RPC. A level overview of some
+of the most important RPC implementation is as follows.
+
+#### Java Remote Method Invocation
+
+Java RMI (Java Remote Method Invocation){% cite rmibook --file rpc %} is a Java
+implementation for performing RPC (Remote Procedure Calls) between a client and
+a server.  The client using a stub passes via a socket connection the
+information over the network to the server.  The Remote Object Registry (ROR){%
+cite rmipaper --file rpc %} on the server contains the references to objects
+that can be accessed remotely and through which the client will connect to.  The
+client then can request of the invocation of methods on the server for
+processing the requested call and then responds with the answer.  RMI provides
+some security by being encoded but not encrypted, though that can be augmented
+by tunneling over a secure connection or other methods.
+
+#### CORBA
+
+CORBA (Common Object Request Broker Architecture){% cite corba --file rpc %} was
+created by the Object Management Group {% cite corbasite --file rpc %} to allow
+for language-agnostic communication among multiple computers.  It is an
+object-oriented model defined via an Interface Definition Language (IDL) and the
+communication is managed through an Object Request Broker (ORB).  Each client
+and server have an ORB by which they communicate.  The benefits of CORBA is that
+it allows for multi-language implementations that can communicate with each
+other, but much of the criticism around CORBA relates to poor consistency among
+implementations.
+
+#### XML-RPC and SOAP
+
+SOAP (Simple Object Access Protocol) is a successor of XML-RPC as a web-services
+protocol for communicating between a client and server. It was initially
+designed by a group at Microsoft {% cite soaparticle1 --file rpc %}.  The SOAP
+message is an XML-formatted message composed of an envelope inside which a
+header and a body are provided.  The body of the message contains the request
+and response of the message, which is transmitted over HTTP or SMTP.  The
+benefit of such a protocol is that it provides the flexibility for transmission
+over multiple transport protocol, though parsing such messages could become a
+bottleneck.
+
+#### Thrift
+
+Thrift is an RPC system created by Facebook and now part of the Apache
+Foundation {% cite thrift --file rpc %}. It is a language-agnostic IDL by which
```

Comment:
@MuzammilAR  Please add the following:

A Interface Description Language (IDL) is a language for specifying a common protocol for interfacing among API frameworks. {% cite OMG-IDL-Syntax-and-Semantics %}

Reference

[OMG-IDL-Syntax-and-Semantics] http://www.omg.org/cgi-bin/doc?formal/02-06-39.pdf